### ISSUE TITLE
Crashlytics Xcode compile warnings: 'swift_name' attribute argument must be a string literal specifying a Swift function name

### ISSUE THREAD

**Issue by mikehardy** [firebase/firebase-ios-sdk#9331]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 13.2.1
  * Firebase SDK version: 8.12.1
  * Installation method: `CocoaPods` (select one)
  * Firebase Component: crashlytics headers (Auth, Core, Database, Firestore, Messaging, Storage, etc)
  * Target platform(s): `iOS`

#### Steps to reproduce:

Compile some part of firebase-ios-sdk that pulls in crashlytics, and as Xcode chews on the header files it spits out two warnings:

```

⚠️  /Users/runner/work/react-native-firebase/react-native-firebase/tests/ios/Pods/Headers/Private/FirebaseCrashlytics/FIRCrashlyticsReport.h:72:37: 'swift_name' attribute argument must be a string literal specifying a Swift function name [-Wswift-name-attribute]
            arguments:(va_list)args NS_SWIFT_NAME(log(format:arguments:));
                                    ^
⚠️  /Users/runner/work/react-native-firebase/react-native-firebase/tests/ios/Pods/Headers/Private/FirebaseCrashlytics/FIRCrashlytics.h:75:37: 'swift_name' attribute argument must be a string literal specifying a Swift function name [-Wswift-name-attribute]
            arguments:(va_list)args NS_SWIFT_NAME(log(format:arguments:));
                                    ^
```

(here it is in the react-native-firebase build system: https://github.com/invertase/react-native-firebase/runs/5143899791?check_suite_focus=true#step:18:2204)

#### Relevant Code:

I think you just need to include crashlytics and start attempting to use it in any way, but as linked above you can see it when compiling the react-native-firebase e2e test app:

- clone react-native-firebase repo
- yarn
- yarn tests:ios:pod:install
- yarn tests:ios:build

Not the highest priority item in the world but I just swept through the module and tried to make our build quiet, so now these stick out like a sore thumb, thought you all might be interested

https://github.com/invertase/react-native-firebase/pull/6061

**Comment by paulb777** [comment]
@ncooke3 found that this is likely a duplicate of #4851 and has some ideas about alternative syntax so the preprocessor doesn't change the `log` symbol.

**Comment by ncooke3** [comment]
Hi @mikehardy, thanks for reporting. Like Paul mentioned above, the issue was due to a non-Firebase defined macro that altered a Crashlytics header. A fix is staged for the 9.1.0 release that should address this warning!
